File: thousands-tv/wildcard/packages/interfaces/src/db/interfaces/claimedTicket.ts
import { Types } from "mongoose";

export const CLAIMED_TICKETS_TABLE_NAME = "claimed-tickets";

export enum TicketTierType {
    GENERAL_ADMISSION = "general-admission",
    VIP = "vip",
    PREMIUM = "premium",
    WILDPASS = "wildpass",
}

export interface IClaimedTicket {
    userId: Types.ObjectId;
    eventId: string;
    tier: TicketTierType; // ticket type
    organizationId?: Types.ObjectId; // organization associated with the ticket
    creditMultiplier: number; // credit multiplier associated with the ticket
    accessCodeId?: Types.ObjectId; // access code used to claim the ticket (we can use this to track the access code used to claim the ticket)

    // mongo fields
    createdAt?: Date;
    updatedAt?: Date;
    _id?: Types.ObjectId;
    __v?: number;
}

File: thousands-services/IvsIdleGameShared/Models/Ticket/ClaimedTicket.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.InteropServices.JavaScript;
using System.Text;
using System.Threading.Tasks;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace IvsIdleGameShared.Models.Ticket
{
    [BsonIgnoreExtraElements]
    public class ClaimedTicket
    {
        [BsonId]
        [BsonElement("_id")]
        public ObjectId Id { get; set; }

        [BsonElement("userId")]
        public ObjectId UserId { get; set; }

        [BsonElement("eventId")] 
        public string EventId { get; set; } = "";

        [BsonElement("tier")] 
        public string Tier { get; set; } = "";

        [BsonElement("organizationId")]
        public ObjectId? OrganizationId { get; set; } // organization associated with the ticket

        [BsonElement("creditMultiplier")]
        public int CreditMultiplier { get; set; } = 1; // credit multiplier associated with the ticket

        [BsonElement("accessCodeId")]
        public ObjectId? AccessCodeId { get; set; }

        [BsonElement("createdAt")]
        public DateTime? CreatedAt { get; set; }

        [BsonElement("updatedAt")]
        public DateTime? UpdatedAt { get; set; }

        [BsonElement("__v")]
        public int? Version { get; set; }
    }
}
